File: Q1.py
def su(n,sum):
    try:
        k=int(n)
        while k<0:
            print("The input is negative")
            k = int(input("Please enter a non-negative integer "))
        for val in range(1,k+1):
            sum = sum+val
        print("The sum is",sum)
    except ValueError:
        c=input("Error: The number you have entered is not an integer. Try again: ")
        su(c,0)
b = input("Enter the last number of the sum: ")
su(b,0)

    

File: Q1.cpp
#include<iostream>
using namespace std;
bool ion(float n){
    while(n>=0){
        if(n==0)return true;
        n--;
    }
    return false;
}
int main(){
    float c=0,n;
    cout<<"Enter the last number of the sum:";
    cin>>n;
    while(ion(n)==false){
        cout<<"Please enter a non-negative integer \n";
        cout<<"Enter the last number of the sum:";
        cin>>n;
    }
        for(int i=1;i<=n;i++)
    {
        c+=i;
    }
    cout<<"The sum is "<<c;
    }
    

